Add support for MacOS X (10.10)

ankon opened the pull request:
- xargs -r: GNU extension, reimplemented using a shell function
- sed -z: GNU extension, reimplemented by piping through echo (removing
  the newlines) and tr (adding the ',')
- xargs -I{}: reimplemented using a for-loop, to avoid hitting
  the 255-byte limitation for eacho argument of xargs(1)

While there: replace the grep-parsing by a filtering for-loop, which
as a side-effect fixes command line arguments with quotes due to the use
of eval.

ankon committed:
Add support for MacOS X (10.10)

* xargs -r: GNU extension, reimplemented using a shell function
* sed -z: GNU extension, reimplemented by piping through echo (removing
  the newlines) and tr (adding the ',')
* xargs -I{}: reimplemented using a for-loop, to avoid hitting
  the 255-byte limitation for eacho argument of xargs(1)

While there: replace the grep-parsing by a filtering for-loop, which
as a side-effect fixes command line arguments with quotes due to the use
of eval.

Stephan202 commented:
You basically rewrote it ;)

Good point on using `tr(1)`. Will review the other changes later today and merge. Really a shame that cross-platform scripts always seem to boil down to reinventing the wheel :(.

ankon committed:
Skip empty lines

ankon committed:
Avoid a sub-shell: we can just process the output with read

This should improve the speed a bit, and also avoids issues with
files-with-spaces

ankon committed:
Avoid one eval: just use "$@" at the right point

ankon commented:
"Hammered it into working in my environment" :)

Stephan202 commented:
Alright, added some notes. One last thing: we now have lower and upper-case variable names, as well as names with and without a leading underscore. Shall we settle on one style?
